POEM_003.md

```diff
@@ -0,0 +1,152 @@
+POEM ID: 003  
+Title: Adding a pre_setup method to create objects and communicators  
```

Comment:
Need to update the title of this poem now. Suggest "Allowing addition if I/O during Configure"

Reply:
Makes sense. Just changed it